abdm_integrator/hiu/views/health_information.py

```diff
@@ -0,0 +1,295 @@
+        data = {'careContextReference': entry['careContextReference']}
+        decrypted_data_str = hiu_crypto.decrypt(encrypted_data, self.request_data['keyMaterial'])
+        # DISABLING for external HIP TEST
+        # if not hiu_crypto.generate_checksum(decrypted_data_str) == entry['checksum']:
+        #     raise HealthDataReceiverException('Error occurred while decryption process: Checksum failed')
+        data['content'] = json.loads(decrypted_data_str)
+        return data
+
+    def save_health_data_receipt(self, care_contexts_status):
+        HealthDataReceiver.objects.create(health_information_request=self.health_information_request,
+                                          page_number=self.request_data['pageNumber'],
+                                          care_contexts_status=care_contexts_status)
+
+    def generate_care_contexts_status(self, care_contexts, error=None):
+        hi_status = HealthInformationStatus.ERRORED if error else HealthInformationStatus.OK
+        description = error if error else 'Delivered'
+        return [{'careContextReference': care_context['care_context_reference'], 'hiStatus': hi_status,
+                 'description': description} for care_context in care_contexts]
+
+    def set_response_in_cache(self, error=None):
+        if error:
+            self.request_data['error'] = {'code': HIUError.CODE_HEALTH_DATA_RECEIVER,
+                                          'message': HIUError.CUSTOM_ERRORS[HIUError.CODE_HEALTH_DATA_RECEIVER]}
+        cache_key = f"{self.health_information_request.gateway_request_id}_{self.request_data['pageNumber']}"
+        ABDMCache.set(cache_key, self.request_data, HEALTH_DATA_CACHE_TIMEOUT)
+
+    def get_overall_status(self, current_care_context_status):
+        all_care_context_status = current_care_context_status
+        if self.request_data['pageNumber'] > 1:
+            all_care_context_status_lists = self.health_information_request.health_data_receipts.all().values_list(
+                'care_contexts_status', flat=True)
+            all_care_context_status = list(itertools.chain.from_iterable(all_care_context_status_lists))
+        transfer_status = not any(status['hiStatus'] == HealthInformationStatus.ERRORED
+                                  for status in all_care_context_status)
+        return transfer_status, all_care_context_status
+
+    def gateway_health_information_on_transfer(self, transfer_status, care_contexts_status):
+        artefact = self.health_information_request.consent_artefact
+        payload = ABDMRequestHelper.common_request_data()
+        payload['notification'] = {'consent_id': str(artefact.artefact_id),
+                                   'transaction_id': self.request_data['transactionId'],
+                                   'doneAt': datetime.utcnow().isoformat()}
+        payload['notification']['notifier'] = {'type': RequesterType.HIU, 'id': self.get_hiu_id()}
```

Comment:
```suggestion
        payload['notification'] = {
                'consent_id': str(artefact.artefact_id),
                'transaction_id': self.request_data['transactionId'],
                'doneAt': datetime.utcnow().isoformat()},
                'notification': {
                        'notifier': {
                                'type': RequesterType.HIU,
                                'id': self.get_hiu_id()
                        }
                }
        }
```

Reply:
Addressed in ce8722ad9a202048ac5f6c11a52043ae96db3fbd